catalyst/utils/mixup.py

```diff
@@ -0,0 +1,47 @@
+from typing import Dict, List
+
+import numpy as np
+import torch
+
+
+def mixup_batch(
+    batch: Dict[str, torch.Tensor], keys: List[str], alpha: float = 0.2, mode: str = "replace"
+) -> Dict[str, torch.Tensor]:
+    """
+
+    Args:
+        batch: batch to which you want to apply augmentation
+        keys: batch keys to which you want to apply augmentation
+        alpha: beta distribution a=b parameters. Must be >=0. The more alpha closer to zero the
+            less effect of the mixup.
+        mode: mode determines the method of use. Must be in ["replace", "add"]. If "replace"
+            then replaces the batch with a mixed one, while the batch size is not changed
+            If "add", concatenates mixed examples to the current ones, the batch size increases
+            by 2 times.
+
+    Returns:
+        augmented batch
+
+    """
+    assert isinstance(keys, list), f"keys must be list[str], get: {type(keys)}"
```

Comment:
I'd appreciate it if we could check not for `list`s only, but for other `iterable`s too, for example, `tuple`s